### ISSUE TITLE
C# -> VB: generic static classes are converted to modules

### ISSUE THREAD

**Issue by Alexilia** [icsharpcode/CodeConverter#451]
### Input code
```csharp
public static class TestClass<T> where T : Class1, new() {
        static Task task;
        static TestClass() {
        }
        public static Task Method() {
            return task;
        }
    }
```

### Erroneous output
```vbnet
Public Module TestClass(Of T As {Class1, New})
    Private task As Task

    Sub New()
    End Sub

    Public Function Method() As Task
        Return task
    End Function
End Class
```

### Expected output
```vbnet
Public NotInheritable Class TestClass(Of T As {Class1, New})
    Private Shared task As Task

    Shared Sub New()
    End Sub

    Public Shared Function Method() As Task
        Return task
    End Function
End Class
```

### Details
there is compilation error because modules can't be generic

**Comment by GrahamTheCoder** [comment]
We should probably convert it to a Class and make all the members Shared to match the behaviour in this case.
We shouldn't do that for all Modules though since it'd break extension methods. Fortunately extension methods aren't allowed in generic classes.

**Comment by GrahamTheCoder** [comment]
Fixed in #456
